dds/DCPS/DataReaderImpl_T.h

```diff
@@ -975,6 +988,15 @@ namespace OpenDDS {
                              OpenDDS::DCPS::MarshalingType marshaling_type)
   {
     unique_ptr<MessageTypeWithAllocator> data(new (*data_allocator()) MessageTypeWithAllocator);
+
+    if (marshal_skip_serialize_) {
+      if(!MarshalTraitsType::from_message_block(static_cast<MessageType&>(*data), static_cast<ACE_Message_Block&>(*sample.sample_))) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: DataReaderImpl::dds_demarshal: ")
+                   ACE_TEXT("attempting to skip serialize but bad from_message_block.\n")));
+      }
```

Comment:
failed case is proceeding to `store_instance_data`, is that what we want?

Reply:
I think a return makes sense after printing the error to follow the precedent for the rest of `dds_demarshal`.